src/Type/Constant/ConstantArrayType.php

```diff
@@ -172,6 +172,18 @@ public function equals(Type $type): bool
 
 	public function isCallable(): TrinaryLogic
 	{
+		if (count($this->valueTypes) === 2) {
```

Comment:
Support for this use-case in ConstantArrayType needs more work.

`isCallable` and `getCallableParametersAcceptors` need to work in sync. This is currently implemented with private `findTypeAndMethodName` method. But this method only works around single type and method.

I currently don't have any idea how this should be implemented as a `ParametersAcceptor` instance. We have some support for this with `UnionTypeMethodReflection` but that's a little bit different use-case.

With `UnionTypeMethodReflection`, we're calling method `foo()` on two different types.

But here in the constant array, we're calling two different methods on a single type...

Reply:
Let me know if you have any suggestions, otherwise I'll have to close this PR. Thanks.

Reply:
Would it be possible to refactor `ConstantArrayType::getCallableParametersAcceptors` to deal with `ConstantArrayTypeAndMethod[]` by just merging single `ConstantArrayTypeAndMethod::getCallableParametersAcceptors` results together ?

Reply:
eg
```php
public function getCallableParametersAcceptors(): array
{
    return array_merge([], ...array_map(
        function (ConstantArrayTypeAndMethod $typeAndMethodName): array {
            $this->getTypeAndMethodNameParametersAcceptors($typeAndMethodName);
        }, $this->findTypesAndMethodNames()
    );
}
```

where `getTypeAndMethodNameParametersAcceptors` is almost the existing `getCallableParametersAcceptors` method.

Reply:
Hi, I worry it's not that easy. As I already said:

> With UnionTypeMethodReflection, we're calling method foo() on two different types.

> But here in the constant array, we're calling two different methods on a single type...